File: src/MrHotel/WebApp/src/app/services/guest-manager/data/guest.ts
import { Guid } from "@customTypes/.";

export class Guest {
  public constructor(
    public readonly id: Guid,
    public fullName: string,
    public phoneNumber: string,
    public dateOfBirth: Date,
  ) {}

  public static fromDto(dto: GuestDto): Guest {
    return new Guest(
      dto.id,
      dto.fullName,
      dto.phoneNumber,
      new Date(dto.dateOfBirth),
    );
  }
}

export type GuestDto = Omit<Guest, "dateOfBirth"> & {
  dateOfBirth: string;
};

File: src/MrHotel/Database/Entities/Guests/Guest.cs
namespace MrHotel.Database.Entities.Guests;

public class Guest
{
    public required Guid Id { get; set; }

    public required string FullName { get; set; }

    public required string PhoneNumber { get; set; }

    public required DateTimeOffset DateOfBirth { get; set; }
}
